pkg/controller/direct/firestore/firestoredatabase_controller.go

```diff
@@ -190,25 +188,35 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 	}
 	desired.Name = fqn
 
-	// Apply default values.
-	ApplyFirestoreDatabaseDefaults(desired)
+	// Simulate server-side defaulting, so we don't issue updates for fields that
+	// are different only because of defaulting.
+	expected := direct.ProtoClone(desired)
```

Comment:
Awesome! I forgot we have the ProneClone. I was using `proto.Clone` directly but hit a panic when the proto contains unexported fields.